aidial_adapter_vertexai/chat_completion.py

```diff
@@ -28,26 +30,33 @@ def __init__(self, region: str, project_id: str):
 
     @dial_exception_decorator
     async def chat_completion(self, request: Request, response: Response):
+        headers = request.headers
         model: ChatCompletionAdapter = await get_chat_completion_model(
             deployment=ChatCompletionDeployment(request.deployment_id),
             project_id=self.project_id,
             location=self.region,
+            headers=headers,
         )
 
         params = ModelParameters.create(request)
         prompt = await model.parse_prompt(request.messages)
 
         discarded_messages_count = 0
-        if params.max_prompt_tokens is not None:
+        if params.max_prompt_tokens is not None and not isinstance(
+            prompt, UserError
+        ):
             prompt, discarded_messages_count = await model.truncate_prompt(
                 prompt, params.max_prompt_tokens
             )
 
         async def generate_response(usage: TokenUsage, choice_idx: int) -> None:
             with response.create_choice() as choice:
                 consumer = ChoiceConsumer(choice)
-                await model.chat(params, consumer, prompt)
-                usage.accumulate(consumer.usage)
+                if isinstance(prompt, UserError):
```

Comment:
I think it's better to fail fast (after line 42) and avoid further checks in the code.

Reply:
Fixed